components/esp_modem/examples/modem_console/main/modem_console_main.cpp

```diff
@@ -82,6 +117,17 @@ extern "C" void app_main(void)
     
     assert(dce != nullptr);
 
+    if(dte_config.uart_config.flow_control == ESP_MODEM_FLOW_CONTROL_HW)
+    {
+        if (command_result::OK != dce->set_flow_control(2, 2)) {
+            ESP_LOGE(TAG, "Failed to set the set_flow_control mode");
+            return;
+        }
+        ESP_LOGI(TAG, "set_flow_control OK");
+    } else {
+        ESP_LOGI(TAG, "not set_flow_control, because 2-wire mode active.");
```

Comment:
Nitpick: I would remove the `else {}` statement here, as majority of users won't setup the flow-control so this additional message might be misleading.